src/context/toastContext/ToastContext.tsx

```diff
@@ -0,0 +1,81 @@
+import { SnackbarCloseReason } from "@mui/material";
+import { createContext, ReactNode, useContext, useState } from "react";
+import Toast, {
+  ToastSeverity,
+  ToastVariant,
+} from "../../components/toast/Toast";
+
+type ToastContextType = {
+  toastHandler: (
+    variant: ToastVariant,
+    severity: ToastSeverity,
+    message: string
+  ) => void;
+};
+
+type ToastType = {
+  open: boolean;
+  message: string;
+  severity: ToastSeverity | undefined;
```

Comment:
Is this an optional param?
Maybe we can have it like this?
```suggestion
  severity?: ToastSeverity;
```